File: 2036.py
count = int(input())
arrayMinus = []
arrayPlus = []
total = 0

for i in range(0, count):
  inputNumber = int(input())
  if inputNumber == 1:
    total = total + 1
  else:
    if inputNumber > 0:
      arrayPlus.append(inputNumber)
    else:
      arrayMinus.append(inputNumber)

arrayMinus.sort()
arrayPlus.sort(reverse = True)

for i in range(int(len(arrayMinus) / 2)):
  total += arrayMinus[i*2] * arrayMinus[i*2+1]
  
if len(arrayMinus) % 2 == 1:
  total += arrayMinus[len(arrayMinus) - 1]

for i in range(int(len(arrayPlus)/2)):
  total += arrayPlus[i*2] * arrayPlus[i*2+1]

if len(arrayPlus) % 2 == 1:
  total += arrayPlus[len(arrayPlus) - 1]

print(total)

File: 2036.js
let fs = require('fs');
let input = fs.readFileSync('/dev/stdin').toString().split('\n');

let count = input[0];
let arrayMinus = [];
let arrayPlus = [];
let total = 0;

for (let i = 1; i < input.length; i++) {
  if (input[i] !== '') {
    if(input[i] == 1) total += 1;
    else {
      if(input[i] > 0) {
        arrayPlus.push(Number(input[i]));
      } else {
        arrayMinus.push(Number(input[i]));
      }
    }
  }
}

arrayMinus.sort((a, b) => a - b);
arrayPlus.sort((a, b) => b - a);

for(let i = 0; i < arrayMinus.length; i++) {
  if(i+1 == arrayMinus.length) {
    total += Number(arrayMinus[i]);
  } else {
    total += arrayMinus[i] * arrayMinus[i+1];
    i++
  }
}

for(let i = 0; i < arrayPlus.length; i++) {
  if(i+1 == arrayPlus.length) {
    total += Number(arrayPlus[i]);
  } else {
    total += arrayPlus[i] * arrayPlus[i+1];
    i++
  }
}

console.log(total)